snakemake_executor_plugin_slurm/partitions.py

```
@@ -113,6 +130,33 @@ def query_scontrol_partitions(cluster=None) -> str:
+    """
+    Query the default partition name using sinfo.
+
+    A partition marked with an asterisk is considered default.
+    """
+    cmd = "sinfo -sa -o %P"
+    if cluster:
+        cmd += f" -M {cluster}"
+
+    cmd = shlex.split(cmd)
+    try:
+        result = subprocess.run(cmd, capture_output=True, text=True, check=True)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)